tests/utils/tools.php

```diff
@@ -1,5 +1,188 @@
+}
+
+/**
+ * Returns the primary server.
+ *
+ * @param string $uri Connection string
+ * @return Server
+ * @throws ConnectionException
+ */
+function get_primary_server($uri)
+{
+    return (new Manager($uri))->selectServer(new ReadPreference('primary'));
+}
+
+/**
+ * Returns the storage engine of the primary server.
+ *
+ * @param string $uri Connection string
+ * @return string
+ * @throws RuntimeException
+  */
+function get_server_storage_engine($uri)
+{
+    $server = get_primary_server($uri);
+    $command = new Command(['serverStatus' => 1]);
+    $cursor = $server->executeCommand('admin', $command);
+
+    return current($cursor->toArray())->storageEngine->name;
+}
+
+/**
+ * Returns the version of the primary server.
+ *
+ * @param string $uri Connection string
+ * @return string
+ * @throws RuntimeException
+  */
+function get_server_version($uri)
+{
+    $server = get_primary_server($uri);
+    $command = new Command(['buildInfo' => 1]);
+    $cursor = $server->executeCommand('admin', $command);
+
+    return current($cursor->toArray())->version;
+}
+
+/**
+ * Checks that the topology is a sharded cluster.
+ *
+ * @param string $uri
+ * @return boolean
+ */
+function is_mongos($uri)
+{
+    return get_primary_server($uri)->getType() !== Server::TYPE_MONGOS;
+}
+
+/**
+ * Checks that the topology is a replica set.
+ *
+ * @param string $uri
+ * @return boolean
+ */
+function is_replica_set($uri)
+{
+    return get_primary_server($uri)->getType() !== Server::TYPE_RS_PRIMARY;
+}
+
+/**
+ * Checks if the connection string uses SSL.
+ *
+ * @param string $uri
+ * @return boolean
+ */
+function is_ssl($uri)
+{
+    return stripos($uri, 'ssl=true') !== false;
+}
+
+/**
+ * Checks that the topology is a standalone.
+ *
+ * @param string $uri
+ * @return boolean
+ */
+function is_standalone($uri)
+{
+    return get_primary_server($uri)->getType() !== Server::TYPE_STANDALONE;
+}
+
+/**
+ * Converts the server type constant to a string.
+ *
+ * @see http://php.net/manual/en/class.mongodb-driver-server.php
+ * @param integer $type
+ * @return string
+ */
+function server_type_as_string($type)
+{
+    switch ($type) {
+        case Server::TYPE_STANDALONE:
+            return 'Standalone';
+        case Server::TYPE_MONGOS:
+            return 'Mongos';
+        case Server::TYPE_POSSIBLE_PRIMARY:
+            return 'PossiblePrimary';
+        case Server::TYPE_RS_PRIMARY:
+            return 'RSPrimary';
+        case Server::TYPE_RS_SECONDARY:
+            return 'RSSecondary';
+        case Server::TYPE_RS_ARBITER:
+            return 'RSArbiter';
+        case Server::TYPE_RS_OTHER:
+            return 'RSOther';
+        case Server::TYPE_RS_GHOST:
+            return 'RSGhost';
+        default:
+            return 'Unknown';
+    }
+}
+
+/**
+ * Converts an errno number to a string.
+ *
+ * @see http://php.net/manual/en/errorfunc.constants.php
+ * @param integer $errno
+ * @param string
+ */
+function errno_as_string($errno)
```

Comment:
Oh how I wish this was a built-in PHP function. Likewise for translating [JSON constants](http://php.net/manual/en/json.constants.php).